src/main/java/org/takes/rs/RsWithoutHeader.java

```diff
@@ -50,20 +50,23 @@
      */
     public RsWithoutHeader(final Response res, final CharSequence name) {
         super(
+             // @checkstyle AnonInnerLengthCheck (50 lines)
             new Response() {
                 @Override
                 public Iterable<String> head() throws IOException {
-                    final Collection<String> head = new LinkedList<String>();
                     final String prefix = String.format(
                         "%s:", name.toString().toLowerCase(Locale.ENGLISH)
                     );
-                    for (final String hdr : res.head()) {
-                        if (!hdr.toLowerCase(Locale.ENGLISH)
-                            .startsWith(prefix)) {
-                            head.add(hdr);
+                    return new Select<String>(
+                        res.head(),
+                        new Condition<String>() {
+                            @Override
+                            public boolean fits(final String header) {
+                                return !header.toLowerCase(Locale.ENGLISH)
+                                    .startsWith(prefix);
```

Comment:
this condition is repeated in `RqWithoutHeader`. How about separating this piece of code in another class and use it in both cases?